source/slang/slang-ir-layout.cpp

```diff
@@ -376,7 +376,9 @@ static Result _calcSizeAndAlignment(
     default:
         break;
     }
-    if (as<IRResourceTypeBase>(type) || as<IRSamplerStateTypeBase>(type))
+
+    if (as<IRResourceTypeBase>(type) || as<IRSamplerStateTypeBase>(type) ||
+        type->getOp() == kIROp_RaytracingAccelerationStructureType)
```

Comment:
Why is this check here rather than in the switch statement?

Reply:
Because there are many opcodes that are treated as subclass of ResourceTypeBase, and we do not want to duplicate them all in the switch. For example, if we add a new resource type, we don’t want to modify all switches in the code base and add a case for it.

Reply:
I meant specifically the `kIROp_RaytracingAccelerationStructureType` check. Feels weird for it to be grouped into this set of checks instead of handled with the other opcodes.

Reply:
Ah, I see what you mean. I put it here because it logically groups together with other resource types, but putting it in the same cases with the pointer types should also work. I don't have a strong preference here, so I can move it up.

Reply:
done.